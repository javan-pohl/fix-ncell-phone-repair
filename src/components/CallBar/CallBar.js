import React from 'react'
import PropTypes from 'prop-types'
import { callBar } from './CallBar.module.css'
import formatPhone from '../../functions/formatPhone'

function CallBar({ phone }) {
  const phoneFormatted = formatPhone(phone)
  return (
    <div className={callBar} data-testid="callBar">
      <a href={`tel:${phone}`} data-testid="callBarNum">
        &#9742; Call or Text
        {' '}
        {phoneFormatted}
      </a>
    </div>
  )
}

CallBar.propTypes = {
  phone: PropTypes.string,
}

CallBar.defaultProps = {
  phone: '3034216499',
}

export default CallBar
