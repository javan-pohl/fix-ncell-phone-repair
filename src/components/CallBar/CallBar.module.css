
.callBar {
  background-color: black;
  font-size: 13px;
  text-align: center;
  padding: 4px 0;
}