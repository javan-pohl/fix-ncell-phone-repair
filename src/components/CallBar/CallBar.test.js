import 'jsdom-global/register'
import '@testing-library/jest-dom'
import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import CallBar from './CallBar'

const rawNum = '3034216499'
const formattedNum = '(303)421-6499'

test('CallBar displays the number correctly', () => {
  render(<CallBar phone={rawNum} />)
  expect(screen.getByTestId('callBarNum')).toHaveTextContent(formattedNum)
})

describe('CallBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CallBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should have href attribute "tel:3034216499"', () => {
    const { queryByTestId } = render(<CallBar />)
    const callButton = queryByTestId('callBarNum')
    expect(callButton).toHaveAttribute('href', 'tel:3034216499')
  })
})
