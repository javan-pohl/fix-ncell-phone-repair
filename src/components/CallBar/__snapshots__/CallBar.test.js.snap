// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallBar renders correctly 1`] = `
<div
  className="callBar"
  data-testid="callBar"
>
  <a
    data-testid="callBarNum"
    href="tel:3034216499"
  >
    â˜Ž Call or Text
     
    (303)421-6499
  </a>
</div>
`;
